gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/VerifyApiHostsUseCase.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.api.use_case;
+
+import io.gravitee.apim.core.api.domain_service.VerifyApiHostsDomainService;
+import io.gravitee.rest.api.service.common.GraviteeContext;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class VerifyApiHostsUseCase {
+
+    private final VerifyApiHostsDomainService verifyApiHostsDomainService;
+
+    public Output execute(Input input) {
+        if (verifyApiHostsDomainService.checkApiHosts(GraviteeContext.getCurrentEnvironment(), input.apiId(), input.hosts())) {
```

Comment:
⚠️We should not use GraviteeContext in usecases because they can be executed by schedulers and they don't have context attached.
Instead we should give the environmentid as input of the usecase